weave/trace_server/trace_server_converter.py

```diff
@@ -45,8 +47,14 @@ def replace_ref(ref_str: str) -> str:
         return f"{ri.WEAVE_INTERNAL_SCHEME}:///{internal_project_id}/{tail}"
 
     def mapper(obj: B) -> B:
-        if isinstance(obj, str) and obj.startswith(weave_prefix):
-            return typing.cast(B, replace_ref(obj))
+        if isinstance(obj, str):
+            if obj.startswith(weave_prefix):
+                return typing.cast(B, replace_ref(obj))
+            elif obj.startswith(weave_internal_prefix):
+                # It is important to raise here as this would be the result of
+                # an external client attempting to write internal refs directly.
+                # We want to maintain full control over the internal refs.
+                raise ValueError("Encountered internal ref in external data.")
```

Comment:
Is there a reason we might want to obfuscate the reason to raise here? I can't actually think of a good reason, other than, if this only happens when someone is doing something they shouldn't, returning a handy error explaining why doesn't feel right.

Reply:
Hmm - interesting. These will get transformed into 500s i believe, but really should be a 400.

Reply:
ok, modified the tests to assert on status code and removed these messages.